routers/api/v1/repo/file.go

```diff
@@ -362,9 +362,15 @@ func DeleteFile(ctx *context.APIContext, apiOpts api.DeleteFileOptions) {
 	// responses:
 	//   "200":
 	//     "$ref": "#/responses/FileDeleteResponse"
+	//   "400":
+	//     "$ref": "#/responses/error"
+	//   "403":
+	//     "$ref": "#/responses/error"
+	//   "404":
+	//     "$ref": "#/responses/error"
 
 	if !CanWriteFiles(ctx.Repo) {
-		ctx.Error(http.StatusInternalServerError, "DeleteFile", models.ErrUserDoesNotHaveAccessToRepo{
+		ctx.Error(http.StatusForbidden, "DeleteFile", models.ErrUserDoesNotHaveAccessToRepo{
```

Comment:
I hesitate to think this should maybe kept as `StatusInternalServerError` to limit repo discovery.

In case you want to keep `StatusForbidden` you need to also update the others request for consistency. 
Like https://github.com/go-gitea/gitea/pull/10323/files#diff-f94444002d94f8c2766292e69cefbee2R466

Reply:
If you have read only access it should be forbidden status, in case you don't have any access it should be not found status but not internal server error

Reply:
This permission check is actually unnecessary. DeleteFile is checked in api.go

Reply:
Therefore I'm going to remove it as a duplicate.

Reply:
In order to get to DeleteFile you have to pass:

```go
						m.Delete("", bind(api.DeleteFileOptions{}), repo.DeleteFile)
					}, reqRepoWriter(models.UnitTypeCode), reqToken())
				}, reqRepoReader(models.UnitTypeCode))
...
			}, repoAssignment())
```

CanWriteFiles does the following in addition:

```go
// CanWriteFiles returns true if repository is editable and user has proper access level.
func CanWriteFiles(r *context.Repository) bool {
	return r.Permission.CanWrite(models.UnitTypeCode) && !r.Repository.IsMirror && !r.Repository.IsArchived
}
```

Therefore HttpStatusForbidden or HttpBadRequest are the correct statuses not 404 nor 500.